### ISSUE TITLE
wstunnel server specify outgoing traffic address

### ISSUE THREAD

**Issue by ikhwanperwira** [erebe/wstunnel#285]
**Is your feature request related to a problem? Please describe.**
So when client connected to server, the client public IP will following default public IP of server based on *my public ip site* report.

**Describe the solution you'd like**
Instead of default public IP of server like this:
```
wstunnel server ws://172.16.0.2:2052  # the traffic will be forwarded to 10.148.0.69 by default
```

I expect like this:
```
wstunnel server ws://172.16.0.2:2052 --outgoing-traffic-address 172.16.0.2 # any traffic forwarded to this IP, will became forwared to cloudflare public IP.
```

**Describe alternatives you've considered**
I have no idea. None of flag able to solve me.

**Comment by erebe** [comment]
Indeed, your issue is more related to routing than wstunnel itself.
I think using `socket-so-mark` is the correct way to do it.

**Comment by ikhwanperwira** [comment]
If so, then let me try if I have time to programming in rust to make new this feature.

Actually this is possible, inbound interface data will be forwarded to outbound interface.
